cmd/minishift/cmd/openshift/service.go

```diff
@@ -50,25 +53,78 @@ var serviceCmd = &cobra.Command{
 			atexit.ExitWithMessage(1, "You must specify the name of the service.")
 		}
 
+		host, err := api.Load(constants.MachineName)
+		if err != nil {
+			atexit.ExitWithMessage(1, err.Error())
+		}
+
+		util.ExitIfNotRunning(host.Driver, constants.MachineName)
+
+		ip, err := host.Driver.GetIP()
+		if err != nil {
+			atexit.ExitWithMessage(1, fmt.Sprintf("Error getting IP: %s", err.Error()))
+		}
+
 		service := args[0]
 
-		url, err := openshift.GetServiceSpec(service, namespace, https)
+		serviceSpec, err := openshift.GetServiceSpec(service, namespace, https)
 		if err != nil {
 			atexit.ExitWithMessage(1, err.Error())
 		}
 
-		if urlMode {
-			fmt.Fprintln(os.Stdout, url)
+		if url {
```

Comment:
This is not only for integration test, but simply to get the single url to standard out. What you do with it is up to you. The table format is nice to read, but almost impossible to process in a script. That only the first URL is returned is a decision we made to keep things simpler. --url option or not this "issue" exists. If you open in browser you the same "issue". We decided to open the first.